frame/treasury/src/lib.rs

```diff
@@ -165,8 +168,8 @@ decl_module! {
 		/// # </weight>
 		#[weight = SimpleDispatchInfo::FixedOperational(100_000)]
 		fn reject_proposal(origin, #[compact] proposal_id: ProposalIndex) {
-			T::RejectOrigin::ensure_origin(origin)?;
-			let proposal = <Proposals<T>>::take(proposal_id).ok_or("No proposal at that index")?;
+			T::RejectOrigin::ensure_origin(origin).map_err(|e| Into::<&str>::into(e))?;
```

Comment:
should also be using `.as_str()`

Reply:
`ensure_origin` returns the `InvalidOrigin` struct that doesn't have `as_str` method because it doesn't implement `ModuleDispatchError`. It only has `From<InvalidOrigin> for &'static str` implementation. Am I missing something?

Reply:
Then we should probably alter it so there are proper errors for invalid origin errors. Using strings for these kinds of super-well-known global errors seems dumb.

Reply:
(Can go in a separate PR though, I guess)

Reply:
See https://github.com/paritytech/substrate/issues/4409